From e1421d21d95b0e27b018dbb8d4d6fb22b604f0f6 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:03 -0400
Subject: [powerpc] Only print PACATMSCRATCH in oops when TM is active

Message-id: <1410545655-205645-195-git-send-email-dzickus@redhat.com>
Patchwork-id: 94294
O-Subject: [RHEL7 PATCH 194/626] powerpc: Only print PACATMSCRATCH in oops when TM is active
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 6d888d1ab0000dff8ea2901bcdf5d213f2a54e8b
Author: Anton Blanchard <anton@samba.org>
Date:   Mon Nov 18 13:19:17 2013 +1100

    powerpc: Only print PACATMSCRATCH in oops when TM is active

    If TM is not active there is no need to print PACATMSCRATCH
    so we can save ourselves a line.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Acked-by: Michael Neuling <mikey@neuling.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/process.c b/arch/powerpc/kernel/process.c
index ad11fb5..b36e3eb 100644
--- a/arch/powerpc/kernel/process.c
+++ b/arch/powerpc/kernel/process.c
@@ -1015,7 +1015,8 @@ void show_regs(struct pt_regs * regs)
  printk("SOFTE: %ld ", regs->softe);
 #endif
 #ifdef CONFIG_PPC_TRANSACTIONAL_MEM
- printk("\nPACATMSCRATCH: %016llx ", get_paca()->tm_scratch);
+ if (MSR_TM_ACTIVE(regs->msr))
+  printk("\nPACATMSCRATCH: %016llx ", get_paca()->tm_scratch);
 #endif
 
  for (i = 0;  i < 32;  i++) {
-- 
1.7.1